src/wallet/api/wallet.cpp

```diff
@@ -487,6 +490,8 @@ uint64_t WalletImpl::approximateBlockChainHeight() const
 }
 uint64_t WalletImpl::daemonBlockChainHeight() const
 {
+    if (!m_is_connected || m_is_shutting_down)
+        return 0;
```

Comment:
Is returning zero the correct approach? In the error path below, how does the caller know whether 0 is the correct or failure value? This is a public function, and 0 doesn't automatically strike me as an invalid value. This is continuing the existing behavior so ...